src/Resources/ResourceManager/Implementation/Resource/RemoveAzureResourceCmdlet.cs

```diff
@@ -51,6 +54,7 @@ protected override void OnProcessRecord()
                             odataQuery: this.ODataQuery)
                         .Result;
 
+                  if (!NoWait) {
```

Comment:
After talking through the scenarios, I agree with Niander that we should return an AzureOperationResponse here, that would allow the user to poll the operation if they wanted to.  This would mean adding an OutputType to the cmdlet OutputType attribute as well